packages/embark/src/lib/utils/template_generator.js

```diff
@@ -73,52 +74,150 @@ class TemplateGenerator {
-          console.log('-------------------'.yellow);
-          console.log(__('Next steps:').green);
-          console.log(('-> ' + ('cd ' + fspath).bold.cyan).green);
-          console.log('-> '.green + 'embark run'.bold.cyan);
-          console.log(__('For more info go to http://embark.status.im').green);
+
+    const templatePkg = `embark-dapp-template-${this.templateName}`;
+    let templateSpecifier;
+    const monorepoRoot = this.insideMonorepo();
+    if (monorepoRoot) {
+      templateSpecifier = utils.joinPath(
+        monorepoRoot, 'dapps/templates', this.templateName
+      );
+    } else {
+      const version = fs.readJSONSync(fs.embarkPath('package.json')).version;
+      templateSpecifier = `${templatePkg}@^${version}`;
+    }
+
+    const tmpDir = require('fs-extra').mkdtempSync(
+      utils.joinPath(require('os').tmpdir(), `${this.templateName}-`)
+    );
+
+    execSync(`npm pack ${templateSpecifier}`, {cwd: tmpDir, stdio: 'ignore'});
+    const packed = utils.filesMatchingPattern(
+      [utils.joinPath(tmpDir, '*.tgz')]
+    )[0];
+
+    this.extract(packed, fspath, () => {
+      this.installTemplate(
+        fspath,
+        name,
+        true,
+        () => {
+          if (name === 'embark_demo') {
+            console.log('-------------------'.yellow);
+            console.log(__('Next steps:').green);
+            console.log(('-> ' + ('cd ' + fspath).bold.cyan).green);
+            console.log('-> '.green + 'embark run'.bold.cyan);
+            console.log(__('For more info go to http://embark.status.im').green);
+          }
+        }
+      );
+    });
+  }
+
+  insideMonorepo() {
+    if (this._monorepoRoot === undefined) {
+      const maybeInsideMonorepo = fs.existsSync(
+        fs.embarkPath('../../packages/embark')
+      );
+      let monorepoRoot = null;
+      if (maybeInsideMonorepo) {
+        const lernaJsonPath = findUp.sync('lerna.json', {cwd: fs.embarkPath()});
+        if (lernaJsonPath) {
+          monorepoRoot = utils.dirname(lernaJsonPath);
         }
       }
-    );
+      this._monorepoRoot = monorepoRoot;
+    }
+    return this._monorepoRoot;
   }
 
   installTemplate(templatePath, name, installPackages, cb) {
     utils.cd(templatePath);
-    utils.sed('package.json', '%APP_NAME%', name);
+
+    const pkgJson = fs.readJSONSync('./package.json');
+    pkgJson.name = name;
+    if (!(/demo/).test(name)) {
+      pkgJson.version = '0.0.1';
+    }
+    delete pkgJson.files;
+    if (!pkgJson.scripts) pkgJson.scripts = {};
+    delete pkgJson.scripts.ci;
+    delete pkgJson.scripts.clean;
+    delete pkgJson.scripts.package;
+    delete pkgJson.scripts.qa;
+    delete pkgJson.scripts.reset;
+    pkgJson.scripts.test = 'embark test';
+    if (pkgJson.devDependencies) {
+      delete pkgJson.devDependencies['embark'];
+      delete pkgJson.devDependencies['embark-reset'];
+      delete pkgJson.devDependencies['npm-run-all'];
+      delete pkgJson.devDependencies['rimraf'];
+    }
```

Comment:
Why do you alter the package.json?

Reply:
So that templates are hooked into the monorepo scripts, e.g. when running `yarn ci`, `yarn reboot`, etc.

Reply:
Ignore my previous question!!
good one!

Reply:
Nice one!